src/main/java/appeng/client/gui/widgets/GuiAeButton.java

```diff
@@ -29,6 +30,14 @@ public String getMessage() {
         }
     }
 
+    @Override
+    public void drawButton(Minecraft mc, int mouseX, int mouseY) {
+        super.drawButton(mc, mouseX, mouseY);
+        if (height < 20) {
+            drawHorizontalLine(xPosition, xPosition + width - 1, yPosition + height - 1, 0xFF000000);
+        }
+    }
+
```

Comment:
This will add the shadow.

```suggestion
    @Override
    public void drawButton(Minecraft mc, int mouseX, int mouseY) {
        super.drawButton(mc, mouseX, mouseY);
        if (height < 20) {
            int hoverState = this.getHoverState(this.field_146123_n);
            switch (hoverState) {
                case 0:
                    this.drawHorizontalLine(xPosition + 2, xPosition + width - 2, yPosition + height - 2, 0xFF2C2C2C);
                    break;
                case 1:
                    this.drawHorizontalLine(xPosition + 2, xPosition + width - 2, yPosition + height - 2, 0xFF565656);
                    break;
                case 2:
                    this.drawHorizontalLine(xPosition + 2, xPosition + width - 2, yPosition + height - 2, 0xFF5C669D);
                    break;
            }
            this.drawHorizontalLine(xPosition, xPosition + width - 1, yPosition + height - 1, 0xFF000000);
        }
        else super.drawButton(mc, mouseX, mouseY);
    }
```

The alternative is completely overriding the vanilla call so that it does the clever X direction stuff also in the Y direction.